drivers/vmwarevsphere/cloudinit.go

```diff
@@ -227,31 +227,50 @@ func (d *Driver) mountCloudInitIso(vm *object.VirtualMachine, dc *object.Datacen
 	return vm.AddDevice(d.getCtx(), add...)
 }
 
-func addSSHUserToYaml(yamlcontent, user, sshkey string) (string, error) {
+func addSSHUserToYaml(yamlcontent, user, group, sshkey string) (string, error) {
 	cf := make(map[interface{}]interface{})
 	if err := yaml.Unmarshal([]byte(yamlcontent), &cf); err != nil {
 		return "", err
 	}
 
 	newUser := map[interface{}]interface{}{
-		"name":        user,
-		"sudo":        "ALL=(ALL) NOPASSWD:ALL",
-		"lock_passwd": "true",
+		"name":          user,
+		"sudo":          "ALL=(ALL) NOPASSWD:ALL",
+		"lock_passwd":   true,
+		"groups":        group,
```

Comment:
What spec is this implementing? It might be worth a comment since this is an `interface{}`

Reply:
https://github.com/canonical/cloud-init/blob/master/cloudinit/config/cc_users_groups.py#L28-L71

Reply:
yup, added.